docs: form field label html acceptatiecriteria (#1251)
nl-design-system/candidate: packages/docs/form-field-label-docs/docs/acceptance-criteria.md

Reviewer: @RenateRoke moet hier ook iets bij voor 1.4.1 Use of color voor de disabled variant?

Reviewer 2: @Robbert is dat iets wat wij technisch oplossen met onze component en iemand die het gaat gebruiken niks meer aan hoeft te doen?

Author: Deze kunnen jullie los van mijn PR in de kernteam Slack bespreken: https://codefornl.slack.com/archives/C02CWGL2M18/p1784274342019009

Reviewer 2: @Robbert  Bij criteria voor gebruik staat al 'De kleur van het Form Field Label geeft geen informatie die niet ook op minimaal 1 andere manier wordt gegeven.' 
Mocht je toch iets voor Criteria van de component zelf bedoelen, let me know :) dan maak ik een apart issue aan (staat ook los van de PR van Charlotte)

Rich text (code) / Text - string (design)

## Semantiek

Dit zijn de acceptatiecriteria welke HTML elementen en HTML attributen het beste zijn voor deze component.

### Welke HTML elementen het beste zijn voor deze component

#### Must Have elementen

- De Form Field Label is opgemaakt met het HTML `label` element.

#### Could Have elementen

> - De Form Field Label mag een inhoud bevatten welke bestaat in de vorm van een `p` element en [phrasing content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#phrasing-content-0) met uitzondering van [interactive content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#interactive-content-0).

Reviewer 3: Toch geen p element op basis van [ Candidate gogogo](https://codefornl.slack.com/docs/T68FXPFQV/F0BGYP6KYBT?focus_section_id=temp%3AC%3ACeZ91bb6501e7c340d6bf2e98655) meeting

Author: Verwerkt!

### Welke HTML attributen het beste zijn voor deze component

#### Must Have attributen

> - De Form Field Label bevat altijd een HTML `for` attribuut, zodat het met een HTML `id` attribuut aan het input-element gekoppeld kan worden.

Reviewer 3: Deze optioneel, zie [Candidate gogogo meeting](https://codefornl.slack.com/docs/T68FXPFQV/F0BGYP6KYBT?focus_section_id=temp%3AC%3ACeZ91bb6501e7c340d6bf2e98655)

Author: Done

- De Form Field Label gebruikt geen HTML `tabindex` attribuut, de component is ondersteunend, niet interactief.

### Alternatieve HTML

- De CSS ondersteund een Form Field Label met een BEM block class op de HTML `label` element.
- De CSS ondersteund een Form Field Label met een BEM block class op de HTML `div` element.
- De CSS ondersteund een Form Field Label met een BEM block class op de HTML `span` element.
- De CSS ondersteund een Form Field Label binnen een element met de `nl-html` class met een HTML `label` element zonder component classes.
- De CSS ondersteund een Form Field Label binnen een element met de `nl-html--form-field-label` class met een HTML `label` element zonder component classes.
- De CSS ondersteund een semantische HTML Label bestaande uit een HTML `label` element zonder component classes.

### Forced Colors modus

- De Form Field Label tekstkleur in Forced Colors modus is CanvasText.
- De Form Field Label tekstkleur van de `disabled` variant in Forced Colors modus is GrayText.
- De Form Field Label tekstkleur van de `option` variant in Forced Colors modus is ButtonText.

## Acceptatiecriteria toegankelijkheid van de component

> - Het Form Field Label kan aan een element voor formulierinvoer gekoppeld worden.

Reviewer 3: Niet "kan" maar "moet"? Dat het gekoppeld moet zijn dmv een `for` of dat de label het veld wrapt

Author: Is optioneel geworden volgens de Go Go Go dus 'kan'

Rich text (code) / Text - string (design)

## Semantiek

> Dit zijn de acceptatiecriteria welke HTML elementen en HTML attributen het beste zijn voor deze component.

Reviewer: ```suggestion
Dit zijn de acceptatiecriteria welke HTML-elementen en HTML-attributen het beste zijn voor deze component.
```

Author: Ik maak een issue aan om dit consistent toe te passen door heel candidate repo, gezien de verschillen:
<img width="546" height="553" alt="Screenshot 2026-07-17 at 09 32 33" src="https://github.com/user-attachments/assets/ed7c7fef-b6f8-4496-8d17-cc1c3e2aff08" />
<img width="537" height="458" alt="Screenshot 2026-07-17 at 09 32 29" src="https://github.com/user-attachments/assets/003999d8-ac07-4570-93fd-8049a3f5f125" />
<img width="541" height="772" alt="Screenshot 2026-07-17 at 09 32 15" src="https://github.com/user-attachments/assets/e5c036d5-a383-43b0-b1dc-6324a2534a34" />

<img width="539" height="384" alt="Screenshot 2026-07-17 at 09 32 22" src="https://github.com/user-attachments/assets/b22341e0-4d46-4335-9ee7-129a7cea4c85" />

Author: Issue gemaakt: https://github.com/nl-design-system/candidate/issues/1262

> ### Welke HTML elementen het beste zijn voor deze component

Reviewer: ```suggestion
### Welke HTML-elementen het beste zijn voor deze component
```

Author: Issue gemaakt: https://github.com/nl-design-system/candidate/issues/1262

#### Must Have elementen

> - De Form Field Label is opgemaakt met het HTML `label` element.

Reviewer: ```suggestion
- De Form Field Label is opgemaakt met het HTML-element `label`.
```

Author: Issue gemaakt: https://github.com/nl-design-system/candidate/issues/1262

#### Could Have elementen

- De Form Field Label mag een inhoud bevatten welke bestaat in de vorm van Radio Button, Checkbox of Switch component.
- De Form Field Label mag een inhoud bevatten welke bestaat in de vorm van [phrasing content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#phrasing-content-0) met uitzondering van [interactive content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#interactive-content-0).

> ### Welke HTML attributen het beste zijn voor deze component

Reviewer: ```suggestion
### Welke HTML-attributen het beste zijn voor deze component
```

Author: Issue gemaakt: https://github.com/nl-design-system/candidate/issues/1262

#### Must Have attributen

> - De Form Field Label gebruikt geen HTML `tabindex` attribuut, de component is ondersteunend, niet interactief.

Reviewer: ```suggestion
- De Form Field Label gebruikt geen `tabindex` HTML-attribuut. De component is informatief, niet interactief.
```

#### Could Have attributen

> - De Form Field Label accepteert een HTML `for` attribuut, zodat het met een HTML `id` attribuut aan het input-element gekoppeld kan worden.

Reviewer: ```suggestion
- De Form Field Label accepteert een `for` HTML-attribuut, zodat het met een `id` HTML-attribuut aan het input-element gekoppeld kan worden.
```

Author: Issue gemaakt: https://github.com/nl-design-system/candidate/issues/1262

### Alternatieve HTML

#### Alternatieve HTML voor Form Field Label zonder `option`

- De CSS ondersteund een Form Field Label met de `nl-form-field-label` class op de HTML `label` element.
- De CSS ondersteund een Form Field Label met de `nl-form-field-label` class op de HTML `div` element.
- De CSS ondersteund een Form Field Label met de `nl-form-field-label` class op de HTML `span` element.
- De CSS ondersteund een Form Field Label binnen een element met de `nl-html` class met een HTML `label` element zonder component classes.
- De CSS ondersteund een Form Field Label binnen een element met de `nl-html--form-field-label` class met een HTML `label` element zonder component classes.

> - De CSS ondersteund een semantische HTML Label bestaande uit een HTML `label` element zonder component classes.

Reviewer: Ik denk dat het handig zou zijn om de CSS en HTML implementatie te scheiden:

```suggestion
- De CSS ondersteunt een Form Field Label met de `nl-form-field-label` class op de HTML-element `label`.
- De CSS ondersteunt een Form Field Label met de `nl-form-field-label` class op de HTML-element `div`.
- De CSS ondersteunt een Form Field Label met de `nl-form-field-label` class op de HTML-element `span`.

Voor de HTML component:

- De CSS ondersteunt een Form Field Label binnen een element met de `nl-html` class met een `label` HTML-element zonder component classes.
- De CSS ondersteunt een Form Field Label binnen een element met de `nl-html--form-field-label` class met een `label` HTML-element zonder component classes.
- De CSS ondersteunt een `label` HTML-element, zonder dat er class names nodig zijn.
```

#### Alternatieve HTML voor Form Field Label met `option`

- De CSS ondersteund een Form Field Label met de `nl-form-field-label` en `nl-form-field-label--option` class op de HTML `label` element.
- De CSS ondersteund een Form Field Label met de `nl-form-field-label` en `nl-form-field-label--option` class op de HTML `div` element.
- De CSS ondersteund een Form Field Label met de `nl-form-field-label` en `nl-form-field-label--option` class op de HTML `span` element.
- De CSS ondersteund een Form Field Label binnen een element met de `nl-html` class met een HTML `label` element zonder component classes.
- De CSS ondersteund een Form Field Label binnen een element met de `nl-html--form-field-label` class met een HTML `label` element zonder component classes.

> - De CSS ondersteund een semantische HTML Label bestaande uit een HTML `label` element zonder component classes.

Reviewer: @eslook zou je deze ook nog willen opsplitsen?

Author: Verwerkt!

### Forced Colors modus

- De Form Field Label tekstkleur in Forced Colors modus is CanvasText.
- De Form Field Label tekstkleur van de `disabled` variant in Forced Colors modus is GrayText.

> - De Form Field Label tekstkleur van de `option` variant in Forced Colors modus is CanvasText.

Reviewer: ```suggestion
- De Form Field Label tekstkleur in Forced Colors modus is `CanvasText`.
- De Form Field Label tekstkleur van de `disabled` variant in Forced Colors modus is `GrayText`.
- De Form Field Label tekstkleur van de `option` variant in Forced Colors modus is `CanvasText`.
```

Rich text (code) / Text - string (design)

## Semantiek

Dit zijn de acceptatiecriteria welke HTML elementen en HTML attributen het beste zijn voor deze component.

### Welke HTML elementen het beste zijn voor deze component

#### Must Have elementen

- De Form Field Label is opgemaakt met het HTML `label` element.

#### Could Have elementen

- De Form Field Label mag een inhoud bevatten welke bestaat in de vorm van Radio Button, Checkbox of Switch component.
- De Form Field Label mag een inhoud bevatten welke bestaat in de vorm van [phrasing content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#phrasing-content-0) met uitzondering van [interactive content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#interactive-content-0).

### Welke HTML attributen het beste zijn voor deze component

#### Must Have attributen

> - De Form Field Label gebruikt geen `tabindex` HTML-attribuut. De component is informatief, niet interactief.

Reviewer: ```suggestion
- De Form Field Label gebruikt geen `tabindex` HTML-attribuut. De component is interactief, maar het gekoppelde formuliercomponent moet focusbaar zijn, niet de label.
```

#### Could Have attributen

- De Form Field Label accepteert een HTML `for` attribuut, zodat het met een HTML `id` attribuut aan het input-element gekoppeld kan worden.

### Alternatieve HTML

#### Alternatieve HTML voor Form Field Label zonder `option`

- De CSS ondersteunt een Form Field Label met de `nl-form-field-label` class op de HTML-element `label`.
- De CSS ondersteunt een Form Field Label met de `nl-form-field-label` class op de HTML-element `div`.
- De CSS ondersteunt een Form Field Label met de `nl-form-field-label` class op de HTML-element `span`.

Voor de HTML component:

> - De CSS ondersteunt een Form Field Label binnen een element met de `nl-html` class met een `label` HTML-element zonder component classes.

Reviewer: ```suggestion
- De CSS ondersteunt een Form Field Label binnen een element met de `nl-html--all` class met een `label` HTML-element zonder component classes.
```

- De CSS ondersteunt een Form Field Label binnen een element met de `nl-html--form-field-label` class met een `label` HTML-element zonder component classes.
- De CSS ondersteunt een `label` HTML-element, zonder dat er class names nodig zijn.

> #### Alternatieve HTML voor Form Field Label met `option`

Reviewer: Ik denk dat de `option` variant zo maar op deze manier kan werken. De grote vraag is: hoe gaan we 'm herkennen? Door `label:has(input[type=....])`? Ik dacht dat we al even hadden gespiekt hoe Utrecht het had gedaan.

Author: Utrecht doet het op deze manier https://github.com/nl-design-system/utrecht/blob/main/components/form-label/src/html/_mixin.scss 

```
  input[type="checkbox" i] ~ label,
  label:has(input[type="checkbox" i]) {
  
  input[type="radio" i] ~ label,
  label:has(input[type="radio" i]) {
```

Is dat iets wat je in deze acceptatiecriteria wilt hebben, of als notitie in Ali's issue?

Author: Expliciet controleren per variant: is het mogelijk met de HTML selectors, is dat wenselijk, en als het mogelijk is, op welke manier, zodat we dat vervolgens documenteren in de CSS API in API bepaald.

Author: Zoals besproken, de methode komt in de CSS API terecht zoals we doen bijv hier https://github.com/nl-design-system/candidate/pull/1266/changes#diff-e9ed509099a3c2be8a5b408a1c1baba36f5514805edb3da3d26acfd54068de75R174

De methode is gewenst omdat het in Utrecht beschikbaar is (iedereen gebruikt Utrecht, breid uit op Utrecht, met uitzondering van Amsterdam waar geen HTML implementatie is - [bron API bepaald](https://github.com/nl-design-system/candidate/issues/914#issuecomment-4419562041))

Issue aangemaakt om de stappen te verduidelijken: https://github.com/nl-design-system/candidate/issues/1296

Ik ga in de CSS API PR de oplossing verwerken: https://github.com/nl-design-system/candidate/pull/1250/changes

- De CSS ondersteunt een Form Field Label met de `nl-form-field-label` en `nl-form-field-label--option` class op de HTML `label` element.
- De CSS ondersteunt een Form Field Label met de `nl-form-field-label` en `nl-form-field-label--option` class op de HTML `div` element.
- De CSS ondersteunt een Form Field Label met de `nl-form-field-label` en `nl-form-field-label--option` class op de HTML `span` element.

Voor de HTML component:

- De CSS ondersteunt een Form Field Label binnen een element met de `nl-html` class met een HTML `label` element zonder component classes.
- De CSS ondersteunt een Form Field Label binnen een element met de `nl-html--form-field-label` class met een HTML `label` element zonder component classes.
- De CSS ondersteunt een semantische HTML Label bestaande uit een HTML `label` element zonder component classes.

### Forced Colors modus

- De Form Field Label tekstkleur in Forced Colors modus is `CanvasText`.
- De Form Field Label tekstkleur van de `disabled` variant in Forced Colors modus is `GrayText`.

> - De Form Field Label tekstkleur van de `option` variant in Forced Colors modus is `CanvasText`.

Reviewer: ```suggestion
- De Form Field Label tekstkleur in Forced Colors modus is `currentColor`.
- De Form Field Label tekstkleur van de `disabled` variant in Forced Colors modus is `GrayText`.
- De Form Field Label tekstkleur van de `option` variant in Forced Colors modus is `currentColor`.
```
